internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -165,9 +165,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 	return result, err
 }
 
-func patchMachineDeployment(ctx context.Context, patchHelper *patch.Helper, d *clusterv1.MachineDeployment, options ...patch.Option) error {
+func patchMachineDeployment(ctx context.Context, patchHelper *patch.Helper, md *clusterv1.MachineDeployment, options ...patch.Option) error {
```

Comment:
nit: I would highly recommend to split out a general m=>md rename into a separate PR next time.

Just helps to de-obfuscate the actual changes in this PR and keeps everything nicely separated

(just resolve for this PR)